reconstruct/reconstruct.go

```diff
@@ -101,9 +105,95 @@ func ReconAccuracy(simFilename string, reconFilename string, leavesOnlyFile stri
 	acc := 100 - accuracy
 	answer[allNodes] = acc
 
+	return answer, accByBase
+}
+
+// ReconAccuracyByBase will run if the option given to ReconAccuracy = 1. This function calculates the percentage of first, second and third bases in codons that were correct.
+func ReconAccuracyByBase(simFilename string, reconFilename string, gpFilename string) map[string][]float64 {
+	sim := fasta.Read(simFilename)
+	recon := fasta.Read(reconFilename)
+	genes := genePred.Read(gpFilename)
+	answer := make(map[string][]float64)
+
+	for s := 0; s < len(sim); s++ {
+		for r := 0; r < len(recon); r++ {
+			var percentage1 float64
+			var percentage2 float64
+			var percentage3 float64
+			var mistakes1 float64
+			var mistakes2 float64
+			var mistakes3 float64
+			var total1 float64
+			var total2 float64
+			var total3 float64
+			if sim[s].Name == recon[r].Name {
+				for i := 0; i < len(sim[s].Seq); i++ {
+					for g := 0; g < len(genes); g++ {
+						inExon, exon := simulate.CheckExon(genes[g], i)
+						if inExon {
+							loc := findLocationInCodon(genes[g], exon, i)
+							if loc == 1 {
+								total1 += 1
+								if sim[s].Seq[i] != recon[r].Seq[i] {
+									mistakes1 += 1
+								}
+							} else if loc == 2 {
+								total2 += 1
+								if sim[s].Seq[i] != recon[r].Seq[i] {
+									mistakes2 += 1
+								}
+							} else if loc == 3 {
+								total3 += 1
+								if sim[s].Seq[i] != recon[r].Seq[i] {
+									mistakes3 += 1
+								}
+							}
+						}
+					}
+				}
+				percentage1 = (mistakes1 / total1) * 100
+				percentage2 = (mistakes2 / total2) * 100
+				percentage3 = (mistakes3 / total3) * 100
+				answer[sim[s].Name] = append(answer[sim[s].Name], 100-percentage1)
+				answer[sim[s].Name] = append(answer[sim[s].Name], 100-percentage2)
+				answer[sim[s].Name] = append(answer[sim[s].Name], 100-percentage3)
+			}
+		}
+	}
 	return answer
 }
 
+// findLocationInCodon returns what position in a codon a given base inhabits (first, second or third).
```

Comment:
Consider making this zero-based. This is the same as frame in gtf format which uses zero-based.